src/custom-lists/components/overview/sidebar/Index.js

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { slide as Menu } from 'react-burger-menu'
+import { bindActionCreators } from 'redux'
+
+import { actions, selectors } from 'src/custom-lists'
+import extStyles from './Index.css'
+import MyCollection from './MyCollections'
+import CreateListForm from './CreateListForm'
+import ListItem from './ListItem'
+import DeleteConfirmModal from './DeleteConfirmModal'
+import { actions as filterActs } from 'src/overview/filters'
+
+import { styles } from './ReactBurgerMenu'
+
+class ListContainer extends Component {
+    static propTypes = {
+        getListFromDB: PropTypes.func.isRequired,
+        lists: PropTypes.array.isRequired,
+        handleEditBtnClick: PropTypes.func.isRequired,
+        isDeleteConfShown: PropTypes.bool.isRequired,
+        resetListDeleteModal: PropTypes.func.isRequired,
+        handleCrossBtnClick: PropTypes.func.isRequired,
+        handleListItemClick: PropTypes.func.isRequired,
+        resetFilters: PropTypes.func.isRequired,
+        createPageList: PropTypes.func.isRequired,
+        updateList: PropTypes.func.isRequired,
+        handleAddPageList: PropTypes.func.isRequired,
+        handleDeleteList: PropTypes.func.isRequired,
+        toggleCreateListForm: PropTypes.func.isRequired,
+        showCreateList: PropTypes.bool.isRequired,
+        showCommonNameWarning: PropTypes.bool.isRequired,
+        closeCreateListForm: PropTypes.func.isRequired,
+        setMouseOver: PropTypes.func.isRequired,
+        resetMouseOver: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            listName: null,
+            updatedListName: null,
+            showWarning: false,
+        }
+    }
+
+    async componentWillMount() {
+        // Gets all the list from the DB to populate the sidebar.
+        this.props.getListFromDB()
+    }
+
+    get inputBlockPattern() {
+        return /[^\w\s-]/gi
+    }
+
+    setInputRef = el => (this.inputEl = el)
+
+    handleSearchChange = field => event => {
+        const { value } = event.target
+
+        // Block input of non-words, spaces and hypens for tags
+        if (this.inputBlockPattern.test(value)) {
+            return
+        }
+
+        this.setState(state => ({
+            ...state,
+            [field]: value,
+        }))
+    }
+
+    getSearchVal = value => value.trim().replace(/\s\s+/g, ' ')
+
+    handleCreateListSubmit = event => {
+        event.preventDefault()
+        const { value } = event.target.elements['listName']
+        // value = list name
+        this.props.createPageList(
+            this.getSearchVal(value),
+            this.vacateInputField,
+        )
+    }
+
+    vacateInputField = () => {
+        this.setState({
```

Comment:
Stick to using either the function reducer param version of `setState` (preferred) **or** the object param version. I would really encourage not using both.